source/Subsystem/Yellowstone/Model.cpp

```diff
@@ -0,0 +1,227 @@
+#include "Model.h"
+#include "Vertex.h"
+#include "GLError.h"
+#include "Texture.h"
+
+#define TINYOBJLOADER_IMPLEMENTATION
+#include <tiny_obj_loader.h>
+
+#include <assimp/Importer.hpp>
+#include <assimp/scene.h>
+#include <assimp/postprocess.h>
+
+#include <algorithm>
+#include <unordered_map>
+
+Model::Model(const std::string aPath)
+    : myModelMatrix(1.0f)
+    , myPosition(0.0f)
+    , myVertexArrayObject(0)
+    , myVertexBufferObject(0)
+    , myTextureUnit(0)
+{
+    if (!DoesFileExist(aPath))
+    {
+        printf("%s does not exist\n", aPath.c_str());
+        return;
+    }
+
+    std::size_t lastSlashIndex = aPath.find_last_of("/\\");
+    std::size_t lastDotIndex = aPath.find_last_of(".");
+    std::string baseDirectory = aPath.substr(0, lastSlashIndex + 1);
+    std::string file = aPath.substr(lastSlashIndex + 1, aPath.length() - lastDotIndex);
+    std::string extension = aPath.substr(lastDotIndex + 1, aPath.length());
+
+    if (extension == "obj")
+    {
+        LoadOBJ(aPath, baseDirectory);
+    }
+    else if (extension == "fbx")
+    {
+        LoadFBX(aPath);
+    }
+}
+
+Model::~Model()
+{
+    myMeshes.clear();
+}
+
+bool Model::DoesFileExist(const std::string& aPath)
+{
+    FILE* file;
+    errno_t error = fopen_s(&file, aPath.c_str(), "rb");
+    if (error != 0)
+    {
+        return false;
+    }
+
+    return true;
+}
+
+void Model::LoadOBJ(const std::string aPath, const std::string aBaseDirectory)
```

Comment:
From the looks of things, this Model class is going to be used by a lot of things. Having the load functions for both OBJ and FBX in here and thus copied multiple times in memory, is quite unnecessary. To save some memory, I would suggest moving this type of functionality to a shared class/manager instead so that this function is only present "once".